packages/@vuepress/bundler-vite/src/plugin/createConstantReplacementPlugin.ts

```diff
@@ -0,0 +1,53 @@
+import type { Plugin } from 'vite'
+
+const fileRegex = /\.html\.vue$/
+const templateRegex = /<template>[\s\S]*<\/template>/g
+
+export const createConstantReplacementPlugin = (): Plugin => {
+  let defineRegex: RegExp | null = null
+
+  return {
+    name: 'vuepress:constant-replacement',
+    enforce: 'pre',
+    apply: 'build',
+    configResolved(resolvedConfig) {
+      if (resolvedConfig.define) {
+        const define = Reflect.ownKeys(resolvedConfig.define) as string[]
```

Comment:
Is it necessary to use `Reflect.ownKeys` ?

Reply:
Can be replaced by `Object.keys` if you think that we need. 

`Reflect` is a built-in object, introduced by es6, and it was supported since [node.js v6.0.0](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect#browser_compatibility)

Reply:
Uh uh, `Reflect.ownKeys` also returns non-enumerable prop names or symbols, although this should not be allowed in user define, but we should catch it.

Reply:
- Vite is using spread operator for user defined `define`, which won't allow non-enumerable properties.
- `Object.keys()` has better readability

Reply:
Changed on ef94d6a